bin/elixir

```diff
@@ -159,44 +168,44 @@ while [ $I -le $LENGTH ]; do
     --pipe-to)
         S=3
         RUN_ERL_PIPE="$2"
-        case "$RUN_ERL_PIPE" in
-          -*) echo "--pipe-to : PIPEDIR cannot be a switch" >&2 && exit 1;;
-          *);;
-        esac
-
         RUN_ERL_LOG="$3"
-        case "$RUN_ERL_LOG" in
-          -*) echo "--pipe-to : LOGDIR cannot be a switch" >&2 && exit 1;;
-          *);;
-        esac
+        if [ "$(starts_with "$RUN_ERL_PIPE" "-")" ]; then
+          echo "--pipe-to : PIPEDIR cannot be a switch" >&2 && exit 1
+        elif [ "$(starts_with "$RUN_ERL_LOG" "-")" ]; then
+          echo "--pipe-to : LOGDIR cannot be a switch" >&2 && exit 1
+        fi
         ;;
     --werl)
         if [ "$OS" = "Windows_NT" ]; then ERL_EXEC="werl"; fi
         ;;
     *)
         while [ $I -le $LENGTH ]; do
-          I=$(($I + 1))
+          I=$((I + 1))
           set -- "$@" "$1"
           shift
         done
         break
         ;;
   esac
 
-  I=$(($I + $S))
+  I=$((I + S))
   shift $S
 done
 
-I=$(($E - 1))
+I=$((E - 1))
 while [ $I -ge 0 ]; do
   eval "VAL=\$E$I"
   set -- "$VAL" "$@"
-  I=$(($I - 1))
+  I=$((I - 1))
 done
 
 SELF=$(readlink_f "$0")
 SCRIPT_PATH=$(dirname "$SELF")
 
+# $OSTYPE is not POSIX compliant
+# https://github.com/pstadler/metrics.sh/blob/master/lib/utils/os.sh
+if [ -z "$OSTYPE" ]; then OSTYPE=$(uname | tr '[:upper:]' '[:lower:]'); fi
```

Comment:
I don't think it matters because we are only doing decisions for cygwin and windows, where those are certainly set. :)

Reply:
True. But if we are going to stick with the shellcheck linter, I would recommend to leave it. We could ignore it by passing the error code to the command line as we already do, but if in the future, somebody decides to improve the script and uses the variable it. it will not work in strictly POSIX shells.
Regardless of the linter warnings, by adding this line we can claim our script is POSIX compliant

Reply:
But someone else can set any variable externally, no? how do we handle variables that are set externally then? To be clear, `if [ -n $OSTYPE ] && [ "$OSTYPE" = "cygwin" ]` is not enough?

Reply:
I just checked locally and it is not enough indeed. :(

Reply:
So I checked shellcheck source and skipping 2039 will only skip reserved variables, so i think it is safe to skip it! :+1:

Reply:
this one works too
```
# shellcheck disable=SC2039
if [ -z $OSTYPE ] && [ "$OSTYPE" = "cygwin" ]; then SCRIPT_PATH=$(cygpath -m "$SCRIPT_PATH"); fi
```

Reply:
@josevalim I have pushed a last commit. It's three comments, as we save the trouble of bypassing error codes that may skip useful checks. It think it's a good trade-off